lightning/src/ln/chan_utils.rs

```diff
@@ -858,6 +858,210 @@ pub fn build_anchor_input_witness(funding_key: &PublicKey, funding_sig: &Signatu
+/// - TX outpoint and/or the full Transaction
+/// - confirmation block/height for confirmed transactions
+impl ChannelFundingTransaction {
+	pub(crate) fn new_empty() -> Self {
+		Self { tx: None, txo: None, tx_confirmed_in: None, tx_confirmation_height: None }
+	}
+
+	pub(crate) fn get_tx(&self) -> Option<&Transaction> { self.tx.as_ref() }
+
+	pub(crate) fn set_tx(&mut self, tx: Transaction) {
+		self.tx = Some(tx);
+	}
+
+	pub(crate) fn set_txo(&mut self, txo: chain::transaction::OutPoint) {
+		self.txo = Some(txo);
+	}
+
+	pub(crate) fn is_confirmed(&self) -> bool {
+		self.tx_confirmation_height.is_some() && self.tx_confirmed_in.is_some()
+	}
+
+	/// Returns the current number of confirmations
+	pub(crate) fn get_tx_confirmations(&mut self, height: u32) -> Option<i64> {
+		match self.tx_confirmation_height {
+			// We either haven't seen any confirmation yet, or observed a reorg.
+			None => None,
+			Some(h) => {
+				let confs = height as i64 - h as i64 + 1;
+				// Reset on reorg
+				if confs <= 0 {
+					self.tx_confirmation_height = None;
+				} 
+				Some(confs)
+			}
+		}
+	}
+
+	/// Returns the current number of confirmations
+	pub(crate) fn get_tx_confirmations_nocheck(&self, height: u32) -> Option<u32> {
+		match self.tx_confirmation_height {
+			// We either haven't seen any confirmation yet, or observed a reorg.
+			None => None,
+			Some(h) => height.checked_sub(h).map(|c| c + 1)
+		}
+	}
+
+	fn set_confirmed(&mut self, block_hash: BlockHash, block_height: u32) {
+		// TODO check if confirmed already
+		self.tx_confirmed_in = Some(block_hash);
+		self.tx_confirmation_height = Some(block_height);
+	}
+
+	// fn reset_unconfirmed(&mut self) {
+	// 	self.tx_confirmed_in = None;
+	// 	self.tx_confirmation_height = None;
+	// }
+
+	pub(crate) fn get_tx_confirmed_in(&self) -> Option<BlockHash> { self.tx_confirmed_in }
+	pub(crate) fn get_tx_confirmation_height(&self) -> Option<u32> { self.tx_confirmation_height }
+}
+
+impl Writeable for ChannelFundingTransaction {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		self.tx.write(writer)?;
+		self.txo.write(writer)?;
+		self.tx_confirmed_in.write(writer)?;
+		self.tx_confirmation_height.write(writer)?;
+		Ok(())
+	}
+}
+
+impl Readable for ChannelFundingTransaction {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let tx = Readable::read(reader)?;
+		let txo = Readable::read(reader)?;
+		let tx_confirmed_in = Readable::read(reader)?;
+		let tx_confirmation_height = Readable::read(reader)?;
+		Ok(Self { tx, txo, tx_confirmed_in, tx_confirmation_height })
+	}
+}
+
+/// Stores info about zero or one funding transaction:
+/// - none in case of unfunded channels
+///   1 in case of funded channels (can be uncofirmed or confirmed)
+/// It is prepared to handle multiple funding transaction candidates (e.g. in case of Splicing)
+pub(crate) struct ChannelFundingTransactions {
+	tx: Option<ChannelFundingTransaction>,
+}
```

Comment:
Hmm... naming this one letter apart from `ChannelFundingTransaction` isn't ideal, IMO. How is the interface intended to work with multiple transactions?